pkg/descheduler/pod/pods.go

```diff
@@ -130,6 +133,28 @@ func ListPodsOnANode(
 	return pods, nil
 }
 
+// GetPodOwnerReplicationCount returns the pod owner's replica count
+func GetPodOwnerReplicationCount(ctx context.Context, client clientset.Interface, ownerRef metav1.OwnerReference) (int, error) {
+	switch ownerRef.Kind {
+	case "ReplicaSet":
+		owner, err := client.AppsV1().ReplicaSets(v1.NamespaceAll).Get(ctx, ownerRef.Name, metav1.GetOptions{})
+		if err != nil {
+			return 0, err
+		}
+		return int(owner.Status.Replicas), nil
+	case "ReplicationController":
+		owner, err := client.CoreV1().ReplicationControllers(v1.NamespaceAll).Get(ctx, ownerRef.Name, metav1.GetOptions{})
+		if err != nil {
+			return 0, err
+		}
+		return int(owner.Status.Replicas), nil
+	default:
+		klog.V(4).Infof("pod is non managed by RS or RC - owner name %s kind %s", ownerRef.Name, ownerRef.Kind)
+		// Returning default value as 1 as its a single pod non managed by a rc or rs
+		return 1, nil
+	}
+}
+
 // OwnerRef returns the ownerRefList for the pod.
 func OwnerRef(pod *v1.Pod) []metav1.OwnerReference {
 	return pod.ObjectMeta.GetOwnerReferences()
```

Comment:
hi @ingvagabund  quick question, what does it mean for a pod to have multiple OwnerReferences? is this possible? This method  returns `[]OwnerReference`. Was wondering  if can  assume it will always have length 1, so to avoid a loop. From the test cases it seems there could only be 1.

Reply:
nevermind i found the answer